nototools/android_patches.py

```diff
@@ -32,49 +32,55 @@
 SRC_DIR = tool_utils.resolve_path('[tools]/packages/android')
 DST_DIR = tool_utils.resolve_path('[tools]/packages/android-patched')
 
-def _patch_hyphen_armenian_ethiopic():
-  """Add hyphen-minus glyphs to Armenian and Ethiopic fonts.
+def _patch_hyphen():
+  """Add hyphen-minus glyphs to fonts that need it.
 
-  This is to enable Armenian and Amharic langauges to be hyphenated
-  properly, since Minikin's itemizer currently shows tofus if a an
-  automatically hyphenated word is displated in a font that has neither
-  HYPHEN nor HYPHEN-MINUS.
+  This is to enable languages to be hyphenated properly,
+  since Minikin's itemizer currently shows tofus if an
+  automatically hyphenated word is displated in a font
+  that has neither HYPHEN nor HYPHEN-MINUS.
 
-  In practice only U+002D HYPHEN-MINUS is added, since Noto LGC fonts
+  The list of font names comes from LANG_TO_SCRIPT in
+  tools/font/fontchain_lint.py.
+
+  (In practice only U+002D HYPHEN-MINUS is added, since Noto LGC fonts
   don't have U+2010 HYPHEN.)
 
   Bug: 21570828"""
 
-  FONTS = [
-      'NotoSansArmenian-Regular.ttf',
-      'NotoSansArmenian-Bold.ttf',
-      'NotoSerifArmenian-Regular.ttf',
-      'NotoSerifArmenian-Bold.ttf',
-      'NotoSansEthiopic-Regular.ttf',
-      'NotoSansEthiopic-Bold.ttf',
-  ]
+  # Names of fonts for which Android requires a hyphen.
+  # This list omits Japanese and Korean.
+  script_names = [
+      'Armenian', 'Ethiopic', 'Bengali', 'Gujarati', 'Devanagari',
+      'Kannada', 'Malayalam', 'Oriya', 'Gurmukhi', 'Tamil', 'Telugu']
 
```

Comment:
Just for the record, prime glyphs (googlei18n/noto-fonts#510) were taken care of in Noto Serif.